redwood-widget/src/commonMain/kotlin/app/cash/redwood/widget/Widget.kt

```diff
@@ -38,6 +38,16 @@ public interface Widget<W : Any> {
    */
   public var modifier: Modifier
 
+  /**
+   * Clear event listeners **without** triggering an update the displayed UI.
+   *
+   * After this is called there will be no further updates to properties or children.
+   *
+   * This function is not recursive.
+   */
+  public fun detach() {
```

Comment:
This doesn't appear to ever be called if you aren't using the protocol.

We're also going to force every binding written in Swift to override this.

Reply:
Punting on the non-protocol dispose(). If we’re lucky we can delete the dispose() function soon and never implement it there. https://github.com/cashapp/redwood/issues/2084

Agreed on forcing Swift bindings to implement this. I think it’s fine.